sickbeard/dailysearcher.py

```diff
@@ -84,34 +86,21 @@ def run(self, force=False):
                 # if an error occured assume the episode hasn't aired yet
                 continue
 
+            UpdateWantedList = 0
             ep = show.getEpisode(int(sqlEp["season"]), int(sqlEp["episode"]))
             with ep.lock:
                 if ep.show.paused:
                     ep.status = common.SKIPPED
+                elif ep.season == 0:
+                    logger.log(u"New episode " + ep.prettyName() + " airs today, setting status to SKIPPED because is a special season")
+                    ep.status = common.SKIPPED
                 else:
-                    myDB = db.DBConnection()
-                    sql_selection="SELECT show_name, indexer_id, season, episode, paused FROM (SELECT * FROM tv_shows s,tv_episodes e WHERE s.indexer_id = e.showid) T1 WHERE T1.paused = 0 and T1.episode_id IN (SELECT T2.episode_id FROM tv_episodes T2 WHERE T2.showid = T1.indexer_id and T2.status in (?) ORDER BY T2.season,T2.episode LIMIT 1) and airdate is not null and indexer_id = ? ORDER BY T1.show_name,season,episode"
-                    results = myDB.select(sql_selection, [common.SKIPPED, sqlEp["showid"]])
-                    if not sickbeard.TRAKT_USE_ROLLING_DOWNLOAD:
-                        if ep.season == 0: 
-                            logger.log(u"New episode " + ep.prettyName() + " airs today, setting status to SKIPPED, due to trakt integration")
-                            ep.status = common.SKIPPED
-                        else:  
-                            logger.log(u"New episode " + ep.prettyName() + " airs today, setting status to WANTED")
-                            ep.status = common.WANTED                         
+                    if not sickbeard.TRAKT_USE_ROLLING_DOWNLOAD or not sickbeard.USE_TRAKT:
+                        logger.log(u"New episode " + ep.prettyName() + " airs today, setting status to WANTED")
```

Comment:
Lol, you removed the SQL select which filters if the ep airs today or not, how do you know it airs today? You are setting every episode to wanted before they air ...